cpp/tests/trajectory/test_derive_trajectories.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2020-2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "../base_fixture.hpp"
+
+#include <cuspatial/error.hpp>
+#include <cuspatial/trajectory.hpp>
+
+#include <cudf/column/column.hpp>
+#include <cudf/types.hpp>
+#include <cudf/wrappers/timestamps.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/device_uvector.hpp>
+
+#include <gtest/gtest.h>
+
+struct DeriveTrajectoriesTest : public cuspatial::test::BaseFixture {
+};
+
+TEST_F(DeriveTrajectoriesTest, SizeMismatch)
+{
+  auto const size = 1000;
+
+  {
+    auto id = cudf::column(rmm::device_uvector<int>(size, rmm::cuda_stream_default));
+    auto xs = cudf::column(rmm::device_uvector<float>(size, rmm::cuda_stream_default));
+    auto ys = cudf::column(rmm::device_uvector<float>(size / 2, rmm::cuda_stream_default));
+    auto ts = cudf::column(rmm::device_uvector<cudf::timestamp_ms>(size, rmm::cuda_stream_default));
+
+    EXPECT_THROW(cuspatial::derive_trajectories(id, xs, ys, ts, this->mr()),
+                 cuspatial::logic_error);
+  }
+  {
+    auto id = cudf::column(rmm::device_uvector<int>(size / 2, rmm::cuda_stream_default));
+    auto xs = cudf::column(rmm::device_uvector<float>(size, rmm::cuda_stream_default));
+    auto ys = cudf::column(rmm::device_uvector<float>(size, rmm::cuda_stream_default));
+    auto ts = cudf::column(rmm::device_uvector<cudf::timestamp_ms>(size, rmm::cuda_stream_default));
+    EXPECT_THROW(cuspatial::derive_trajectories(id, xs, ys, ts, this->mr()),
+                 cuspatial::logic_error);
+  }
+  {
+    auto id = cudf::column(rmm::device_uvector<int>(size, rmm::cuda_stream_default));
+    auto xs = cudf::column(rmm::device_uvector<float>(size, rmm::cuda_stream_default));
+    auto ys = cudf::column(rmm::device_uvector<float>(size, rmm::cuda_stream_default));
+    auto ts =
+      cudf::column(rmm::device_uvector<cudf::timestamp_ms>(size / 2, rmm::cuda_stream_default));
+    EXPECT_THROW(cuspatial::derive_trajectories(id, xs, ys, ts, this->mr()),
+                 cuspatial::logic_error);
+  }
```

Comment:
Judging from these tests, I missed the spot above where you are in fact checking that all lengths are the same?

Reply:
We check in the column-based API, but we can't check in the header-only API.

Reply:
Columns have a defined size. Iterators don't.